server/gui/main.py

```diff
@@ -155,14 +180,19 @@ def __init__(self, parent):
         self.signals.selectedFile.connect(self.createScanpyEngine)
 
     def updateEmbedding(self, idx):
-        self.embedding_selection = MODES[idx]
+        self.embedding_selection = [MODES[idx]]
 
     def createScanpyEngine(self, file_name):
-        worker = DataLoadWorker(file_name, self.embedding_selection)
-        worker.signals.result.connect(self.onDataSuccess)
-        worker.signals.error.connect(self.onDataError)
-        self.load_worker = threading.Thread(target=worker.run, daemon=True)
-        self.load_worker.start()
+        worker = Worker(self.window().parent_conn, self.window().child_conn, file_name, self.title, host="127.0.0.1", port=GUI_PORT,
+                        layout=self.embedding_selection)
+        self.window().load_emitter.signals.ready.connect(self.onDataReady)
+        self.window().load_emitter.signals.engine_error.connect(self.onEngineError)
```

Comment:
is there any value in differentiating between engine and server errors?  Ie, will the UI (down the road) need to treat them differently?   If not, might be cleaner to just combine them into one handler.   If they need differentiated UI/handling, fine to keep as is (I just can't think of why we would need that).

Reply:
yes, if the engine fails, it's likely a bad file. If the server fails, I'd eventually want to restart the server with the same file because the error could be just one specific call. And then keep a restart counter so we don't get stuck in an infinite loop of restarts.